src/Infrastructure/GoogleListingsAndAdsPlugin.php

```diff
@@ -50,6 +50,15 @@ public function __construct( ContainerInterface $container ) {
 	public function activate(): void {
 		$this->maybe_register_services();
 
+		// Do not run activate if service registration failed.
```

Comment:
Thanks, @mikkamp, nice catch. I implemented the suggested refactor. And I decided to leave the validation check from maybe_register_services just to be safe.

Reply:
Thanks that gets rid of the double notice. I think it would be nicer to make `$is_activated` private/protected and then refactor activate slightly (I think it's important to break out early):

```php
	public function activate(): void {
		// Delay activation if a required plugin is missing or an incompatible plugin is active.
		if ( ! PluginValidator::validate() ) {
			// Using update_option because we cannot access the option service
			// when the services have not been registered.
			update_option( 'gla_' . OptionsInterface::DELAYED_ACTIVATE, true );
			return;
		}

		$this->maybe_register_services();
		...
```

Reply:
Good catch. It makes things more consistent as well. I have implemented the refactor.